sklbench/benchmarks/sklearn_estimator.py

```diff
@@ -334,34 +334,43 @@ def verify_patching(stream: io.StringIO, function_name) -> bool:
     return acceleration_lines > 0 and fallback_lines == 0
 
 
-def create_online_function(method_instance, data_args, batch_size):
-    n_batches = data_args[0].shape[0] // batch_size
+def create_online_function(
+    estimator_instance, method_instance, data_args, num_batches, batch_size
+):
 
     if "y" in list(inspect.signature(method_instance).parameters):
 
         def ndarray_function(x, y):
-            for i in range(n_batches):
+            for i in range(num_batches):

```

Comment:
Leave old simple logic with `batch_size` only.

Reply:
why?

Reply:
Why change? It overcomplicates data slicing with extra parameter checks and calculations, also, it is more common to know batch size before `partial_fit` call in real world cases.

Reply:
> Why change?

Adding new feature which can be useful.

> It overcomplicates data slicing with extra parameter checks and calculations

It costs nothing. And doing calculations in the code is better than doing them in calculator before running benchmarks.

> it is more common to know batch size before `partial_fit` call in real world cases.

But while doing benchmarking it is not less common (I'd say even more) when the user wants to specify exact number of partial_fit calls.